Task3_Products/crud.py

```diff
@@ -0,0 +1,222 @@
+from typing import Optional, Literal
+
+from fastapi import HTTPException
+from sqlalchemy import asc, desc
+from sqlalchemy.orm import Session
+
+from Task3_Products import models, schemas
+from logger import setup_logger
+
+logger = setup_logger("Shop", "shop.log")
+
+
+def create_category(
+        category_data: schemas.CategoryCreate,
+        db: Session
+) -> models.Category:
+    category_already_exist = db.query(models.Category).filter(
+        models.Category.name == category_data.name).first()
+
+    if category_already_exist:
+        raise HTTPException(status_code=400, detail="Category already exists")
+
+    if category_data.parent_id:
+        parent_category = db.query(models.Category).filter(
+            models.Category.id == category_data.parent_id).first()
+        if not parent_category:
+            raise HTTPException(status_code=404,
+                                detail="Parent category not found")
+    else:
+        category_data.parent_id = None
+
+    category = models.Category(name=category_data.name,
+                               parent_id=category_data.parent_id)
+    db.add(category)
+    db.commit()
+    db.refresh(category)
+    logger.info(f"Category '{category.name}' created")
+    return category
+
+
+def get_categories(
+        db: Session,
+        skip: int = 0,
+        limit: int = 10,
+        sort_by: Literal["id", "name"] = None,
```

Comment:
If it's `None` by default, so top must be `Literal["id", "name"] | None` then

Reply:
Yeah, you're right. I've revised my view on sorting. I think it should be sorted by id by default.